fem/pnonlinearform.cpp

```diff
@@ -217,6 +217,28 @@ void ParBlockNonlinearForm::SetEssentialBC(const
    }
 }
 
+double ParBlockNonlinearForm::GetEnergy(const Vector &x) const
+{
+
+   xs_true.Update(x.GetData(), block_trueOffsets);
+   xs.Update(block_offsets);
+
+   for (int s=0; s<fes.Size(); ++s)
+   {
+      fes[s]->GetProlongationMatrix()->Mult(
+         xs_true.GetBlock(s), xs.GetBlock(s));
+   }
+
```

Comment:
```ParNonlinearForm::GetEnergy``` has the following lines which should probably be added here as well:
```c++
   if (fnfi.Size())
   {
      MFEM_ABORT("TODO: add energy contribution from shared faces");
   }
```

Reply:
GetEnergy calls GetEnergyBlocked which has all these checks.

Reply:
Unit test added!